TestSuite.py

```diff
@@ -23,34 +62,88 @@ def main(args=None):
     dead_or_alive = 0
     alive = 0
 
-    def test(testname: str, command: str):
-        tests[testname] = [command]
-        for mode in [['Open', ''],
-                     ['Std ', ' --mode standard'],
-                     ['Inv ', ' --mode inverted']]:
-
-            basecommand = f"python3.8 DungeonRandomizer.py --door_shuffle {args.dr} --intensity {args.tense} --suppress_rom --suppress_spoiler"
-
-            def gen_seed():
-                taskcommand = basecommand + " " + command + mode[1]
-                return subprocess.run(taskcommand, capture_output=True, shell=True, text=True)
-
-            for x in range(1, max_attempts + 1):
-                task = pool.submit(gen_seed)
-                task.success = False
-                task.name = testname
-                task.mode = mode[0]
-                task.cmd = basecommand + " " + command + mode[1]
-                task_mapping.append(task)
-
-    test("Vanilla   ", "--shuffle vanilla")
-    test("Retro     ", "--retro --shuffle vanilla")
-    test("Keysanity ", "--shuffle vanilla --keydropshuffle --keysanity")
-    test("Shopsanity", "--shuffle vanilla --shopsanity")
-    test("Simple    ", "--shuffle simple")
-    test("Full      ", "--shuffle full")
-    test("Crossed   ", "--shuffle crossed")
-    test("Insanity  ", "--shuffle insanity")
+    def test(testname: list, command: str):
+        tests[' '.join(testname)] = [command]
+        basecommand = f"py -3.8 DungeonRandomizer.py --door_shuffle {args.dr} --intensity {args.tense} --suppress_rom --suppress_spoiler"
+
+        def gen_seed():
+            taskcommand = basecommand + ' ' + command
+            return subprocess.run(taskcommand, capture_output=True, shell=True, text=True)
+
+        for _ in range(1, max_attempts + 1):
+            task = pool.submit(gen_seed)
+            task.success = False
+            task.name = ' '.join(testname)
+            task.settings = testname
+            task.cmd = basecommand + ' ' + command
+            task_mapping.append(task)
+
+    for mode in SETTINGS['mode']:
+        for goal in SETTINGS['goal']:
```

Comment:
Maybe we should add this as an option as a full regression? But this is not practical, I can't run 10-20 seeds for every settings combination. I don't even know how long that would take.